projects/email_alert_notifications/resources/email_alert_notifications.tf

```diff
@@ -0,0 +1,78 @@
+variable "s3_bucket_name" {
+  default = "govuk-email-alert-notifications"
+}
+
+variable "environment"{
+}
+
+variable "rename_email_files_with_request_id_version"{
+}
+
+variable "lambda_bucket"{
+  default = "govuk-lambda-applications"
+}
+
+resource "template_file" "s3_bucket_policy" {
+  template = "${file("templates/email_alert_s3_bucket_policy.json")}"
+  vars {
+    account_id = "${element(split(":", aws_iam_role.lambda_execute_and_write_to_email_alert_bucket.arn), 4)}"
+    bucket_name = "${var.s3_bucket_name}-${var.environment}"
+    lambda_role = "${aws_iam_role.lambda_execute_and_write_to_email_alert_bucket.arn}"
+  }
+}
+
+resource "template_file" "put_and_delete_to_email_alert_bucket_policy" {
+  template = "${file("templates/put_and_delete_to_s3_policy.json")}"
+  vars {
+    resource_arn = "${aws_s3_bucket.email_alert_inbox_bucket.arn}"
+  }
+}
+
+resource "aws_s3_bucket" "email_alert_inbox_bucket" {
+  bucket = "${var.s3_bucket_name}-${var.environment}"
+  acl = "public-read"
+  policy = "${template_file.s3_bucket_policy.rendered}"
+}
+
+resource "aws_iam_role" "lambda_execute_and_write_to_email_alert_bucket" {
+  name = "lambda_execute_and_write_to_email_alert_bucket"
+  assume_role_policy = "${file("templates/lambda_assume_role_policy.json")}"
+}
+
+resource "aws_iam_role_policy" "put_and_delete_to_email_alert_bucket" {
+  name = "put_and_delete_to_email_alert_bucket"
+  role = "${aws_iam_role.lambda_execute_and_write_to_email_alert_bucket.id}"
+  policy = "${template_file.put_and_delete_to_email_alert_bucket_policy.rendered}"
+}
+
+resource "aws_iam_role_policy" "write_to_logs" {
+  name = "write_to_logs"
+  role = "${aws_iam_role.lambda_execute_and_write_to_email_alert_bucket.id}"
+  policy = "${file("templates/write_to_logs_policy.json")}"
+}
+
+resource "aws_lambda_function" "rename_email_files_with_request_id"{
+  s3_bucket = "${var.lambda_bucket}-${var.environment}"
+  s3_key="rename_email_files_with_request_id.zip"
+  s3_object_version="${var.rename_email_files_with_request_id_version}"
```

Comment:
If I'm reading [this correctly](https://github.com/hashicorp/terraform/pull/5370) it suggests this isn't a required setting, and it would make life much easier to pull the latest by default. Makes life more difficult if we wish to deploy a specific version, though.